### PR TITLE
Add escaping for < and > to markdown output

### PR DESCRIPTION
**Pull request by MaximPlusov**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Markdown generation for image descriptions by sanitizing and escaping the alt text during image/picture Markdown rendering.
  * Removed invalid NUL (`\u0000`) characters from alt text instead of leaving them in output.
  * Updated handling of missing/`null` text to emit empty content, with consistent escaping of `&`, `<`, and `>`.
* **Tests**
  * Expanded Markdown generator test coverage for sanitization and escaping edge cases.

### CODE REVIEW

File: java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/markdown/MarkdownGenerator.java [hunk 1]
```diff
@@ -434,10 +434,14 @@ protected void writeSpace() throws IOException {
...
-            return value.replace("\u0000", " ");
+        if (value == null) {
+            return "";
         }
-        return null;
+        return value
+            .replace("\u0000", "")
+            .replace("&", "&amp;")
+            .replace("<", "&lt;")
+            .replace(">", "&gt;");
```
[the remark this answers is not in the document]
Comment (MaximPlusov) [reply java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/markdown/MarkdownGenerator.java:444]: Write test for MarkdownGeneratorTest

**Review by bdoubrov — APPROVED** [review]
